doc/sphinx/changes.rst

```diff
@@ -8,6 +8,753 @@ Recent changes
+  some explicit `Prop` and `SProp` annotations when writing terms.
+  Some minor backwards compatibility issues can arise in rare cases,
+  which can be solved with more explicit sort annotations
+  (`#16903 <https://github.com/coq/coq/pull/16903>`_,
+  by Pierre-Marie Pédrot).
+- **Changed:**
+  match compilation for primitive record avoids producing an encoding overhead for matches that are equivalent to a primitive projection
+  (`#17008 <https://github.com/coq/coq/pull/17008>`_,
+  by Gaëtan Gilbert).
+- **Added:**
+  volatile casts :n:`@term :> @type` which do not leave a trace in the elaborated term.
+  They are used by :flag:`Printing Match All Subterms` to display otherwise hidden subterms of match constructs
+  (`#16992 <https://github.com/coq/coq/pull/16992>`_,
+  fixes `#16918 <https://github.com/coq/coq/issues/16918>`_,
+  by Gaëtan Gilbert).
+- **Added:**
+  when printing uninterpreted terms (for instance through :cmd:`Print Ltac` on `Ltac foo := exact some_term`),
+  extensions to the term language (for instance :ref:`tactics-in-terms`) are now printed correctly instead of as holes (`_`)
+  (`#17221 <https://github.com/coq/coq/pull/17221>`_,
+  by Gaëtan Gilbert).
+- **Added:**
+  Support for the :attr:`local`, :attr:`global` and :attr:`export`
+  locality attributes for the single "field" of :ref:`definitional
+  typeclasses <singleton-class>` when using the ``:>`` and ``::``
+  syntaxes for coercion and substructures
+  (`#17754 <https://github.com/coq/coq/pull/17754>`_,
+  fixes `#17451 <https://github.com/coq/coq/issues/17451>`_,
+  by Pierre Roux).
+- **Fixed:**
+  canonical instance matching `match` terms
+  (`#17206 <https://github.com/coq/coq/pull/17206>`_,
+  fixes `#17079 <https://github.com/coq/coq/issues/17079>`_,
+  by Gaëtan Gilbert).
+- **Fixed:**
+  universe constraint inference in module subtyping can trigger constant unfoldings
+  (`#17305 <https://github.com/coq/coq/pull/17305>`_,
+  fixes `#17303 <https://github.com/coq/coq/issues/17303>`_,
+  by Gaëtan Gilbert).
+
+Notations
+^^^^^^^^^
+
+- **Removed:**
+  The `\'[=\'` keyword. `\'[=\'`
+  tokens in notation definitions should be replaced with the pair of
+  tokens `\'[\' \'=\'`. If compatibility with Coq < 8.18
+  is needed, replace `[=` in uses of the notation with
+  an added space (`[ =`)
+  (`#16788 <https://github.com/coq/coq/pull/16788>`_,
+  fixes `#16785 <https://github.com/coq/coq/issues/16785>`_,
+  by Pierre Roux).
+- **Added:**
+  Support for :flag:`Printing Parentheses` in custom notations
+  (`#17117 <https://github.com/coq/coq/pull/17117>`_, by Hugo
+  Herbelin).
+- **Added:**
+  Improve printing of reverse coercions. When a term :g:`x`
+  is elaborated to :g:`x'` through a reverse coercion,
+  return the term :g:`reverse_coercion x' x` that is convertible
+  to :g:`x'` but displayed :g:`x` thanks to the coercion
+  :g:`reverse_coercion`
+  (`#17484 <https://github.com/coq/coq/pull/17484>`_,
+  by Pierre Roux).
```

Comment:
Is this really about notations or should this be in the previous section? cc @proux01

Reply:
Well, if you consider this is only about displaying reverse coercions, it seems close enough from "Notations"? But indeed if we consider it to be more about coercions in general, then the previous section may be better. No strong opinion, feel free to move it if you prefer.

Reply:
Ok, I'll leave it as is.